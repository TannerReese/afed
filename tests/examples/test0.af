# Tests deep references
# Tests erasure of substitutions
# Tests unnamed map members
# Tests numeric arithmetic

48 / 32 + (100 / 70)^(6/3) = ``,
x  : { 
	2 / 3 - 5/7 = ``,
	67 ^ (2 / 3) * 67^-1 - pi = ``,
	23 % num.e = ``,
y: {
 z
: 
{r
: 78 - 45 / 5 ^
2}}},

answer: 2 * x.y.z.r = `bjnkkm`

