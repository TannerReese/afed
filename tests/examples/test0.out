# Tests deep references
# Tests erasure of substitutions
# Tests unnamed map members
# Tests numeric arithmetic

48 / 32 + (100 / 70)^(6/3) = `347 / 98`,
x  : {
	2 / 3 - 5/7 = `-1 / 21`,
	-  1 + 2 * - 3.1 = `-7.2`,
	67 ^ (2 / 3) * 67^-1 - pi = `-2.8953811166907797`,
	2 ^ 1.1 ^ 1.01 ^ (23 % num.e) = `2.145504449197655`,
	34 // 2.45 % e / 3 * 15 % num.pi = `1.2095854700497188`,
	round (gold ^ 20) = `15127`,
	m: ceil (floor y.z.r / 3) = `26`,
	abs m * signum (1 - m ^ 2) = `-26`,
	sin (atan2 67 32) = `0.9023620172804064`,

	factorial 4 / 39 = `8 / 13`,
	num.choose (sqrt 17) 4 = `1.279358018053861`,
	lcm 21 (gcd 23 69) = `483`,
y: {
 z
:
{r
: 78 - 45 / 5 ^
2}}},

answer: 2 * x.y.z.r = `762 / 5`

